src/main/java/forum/service/ReplyServiceImpl.java

```diff
@@ -0,0 +1,35 @@
+package forum.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import forum.dao.ReplyDAO;
+import forum.entity.Reply;
+import forum.entity.Thread;
+
+@Service
+public class ReplyServiceImpl implements ReplyService {
+	
+	@Autowired
+	private ReplyDAO replyDAO;
+	
+	@Autowired
+	private ThreadService threadService;
+
+	@Override
+	@Transactional
+	public Reply saveReply(Reply reply) {
+		
+		Thread tempThread = threadService.getThread(reply.getThread().getId());
+		
+		tempThread.add(reply);
```

Comment:
How the app behaves if you don't execute this line?

Reply:
This is what the server returns for the POST call when this line is not executed:

```
{
    "id": 3,
    "comment": "A comment for a thread",
    "thread": {
        "id": 67,
        "title": null,
        "content": null,
        "category": null
    }
}
```